File: Swift/ExampleTests/common/RequestTyped.swift
public struct RequestTyped: Equatable, CustomStringConvertible {
    public let method: String
    public let page: String
    public let address: String
    public let benchmark: String
    public let format: String

    public init(method: String, page: String, address: String, benchmark: String, format: String) {
        self.method = method
        self.page = page
        self.address = address
        self.benchmark = benchmark
        self.format = format
    }

    public init(from s: RequestString) {
        self.method = s.method
        self.page = s.page
        self.address = s.address
        self.benchmark = s.benchmark
        self.format = s.format
    }

    public func toStringStruct() -> RequestString {
        return RequestString(
            method: String(describing: method),
            page: String(describing: page),
            address: String(describing: address),
            benchmark: String(describing: benchmark),
            format: String(describing: format)
        )
    }

    public static func toStringList(_ list: [RequestTyped]) -> [RequestString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [RequestString]) -> [RequestTyped] {
        return list.map { RequestTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "method": method,
            "page": page,
            "address": address,
            "benchmark": benchmark,
            "format": format,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.method = try Json.asString(Json.require(m, "method"), "method")
        self.page = try Json.asString(Json.require(m, "page"), "page")
        self.address = try Json.asString(Json.require(m, "address"), "address")
        self.benchmark = try Json.asString(Json.require(m, "benchmark"), "benchmark")
        self.format = try Json.asString(Json.require(m, "format"), "format")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [RequestTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [RequestTyped] {
        return try Json.parseArray(text).map {
            try RequestTyped(fromJSONValue: Json.asObject($0, "RequestTyped"))
        }
    }

    public var description: String {
        return "Method=\(method), Page=\(page), Address=\(address), Benchmark=\(benchmark), Format=\(format)"
    }
}

File: Python/ExampleTests/common/request_typed.py
from . import json_util as _json
from .request_string import RequestString

class RequestTyped:
    def __init__(self, method: str = '', page: str = '', address: str = '', benchmark: str = '', format: str = ''):
        self.method = method
        self.page = page
        self.address = address
        self.benchmark = benchmark
        self.format = format

    @classmethod
    def from_string_obj(cls, s: RequestString) -> 'RequestTyped':
        return cls(
            s.method,
            s.page,
            s.address,
            s.benchmark,
            s.format
        )

    def to_string_obj(self) -> RequestString:
        return RequestString(
            str(self.method),
            str(self.page),
            str(self.address),
            str(self.benchmark),
            str(self.format)
        )

    @staticmethod
    def to_string_list(items) -> list:
        return [t.to_string_obj() for t in items]

    @staticmethod
    def from_string_list(items) -> list:
        return [RequestTyped.from_string_obj(s) for s in items]

    def to_json_value(self) -> dict:
        return {
            'Method': self.method,
            'Page': self.page,
            'Address': self.address,
            'Benchmark': self.benchmark,
            'Format': self.format,
        }

    def to_json(self) -> str:
        return _json.dumps(self.to_json_value())

    @classmethod
    def from_json_value(cls, m: dict) -> 'RequestTyped':
        return cls(
            _json.as_str(_json.require(m, 'Method'), 'Method'),
            _json.as_str(_json.require(m, 'Page'), 'Page'),
            _json.as_str(_json.require(m, 'Address'), 'Address'),
            _json.as_str(_json.require(m, 'Benchmark'), 'Benchmark'),
            _json.as_str(_json.require(m, 'Format'), 'Format')
        )

    @classmethod
    def from_json(cls, text: str) -> 'RequestTyped':
        return cls.from_json_value(_json.loads(text))

    @staticmethod
    def to_json_list(items) -> str:
        return _json.dumps([item.to_json_value() for item in items])

    @classmethod
    def from_json_list(cls, text: str) -> list:
        raw = _json.as_list(_json.loads(text), 'RequestTyped')
        return [cls.from_json_value(e) for e in raw]

    def __str__(self):
        return (f'Method={self.method}' + ', ' +
                f'Page={self.page}' + ', ' +
                f'Address={self.address}' + ', ' +
                f'Benchmark={self.benchmark}' + ', ' +
                f'Format={self.format}')

    def _key(self):
        return (self.method, self.page, self.address, self.benchmark, self.format)

    def __eq__(self, other):
        if not isinstance(other, RequestTyped):
            return NotImplemented
        return self._key() == other._key()

    def __hash__(self):
        return hash(self._key())

    def __repr__(self):
        return f'RequestTyped({self})'
